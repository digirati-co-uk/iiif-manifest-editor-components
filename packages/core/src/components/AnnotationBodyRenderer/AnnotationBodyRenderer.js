import React from 'react';
import { getW3cAnnotationStyle } from '../../utils/IIIFResource';
import { EditorConsumer } from '../EditorContex/EditorContext';

const NotSupportedContentType = (annotation, type) => (
  <p style={getW3cAnnotationStyle(annotation.style || '')}>
    Annotation type[{type}] not supported.
  </p>
);

const AnnotationBodyRenderer = ({ annotation }) => {
  const type = `${annotation.body.type}::${annotation.motivation}`;
  return (
    <EditorConsumer>
      {configuration => {
        if (configuration.annotation.hasOwnProperty(type)) {
          return configuration.annotation[type].contentRenderer(annotation);
        } else {
          return NotSupportedContentType(annotation, type);
        }
      }}
    </EditorConsumer>
  );
};

export default AnnotationBodyRenderer;
