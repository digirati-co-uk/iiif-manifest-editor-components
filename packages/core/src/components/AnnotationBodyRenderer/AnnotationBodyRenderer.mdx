---
name: Annotation Body Renderer
menu: 03. Components
---


# Annotation Body Renderer

The components selects the appropriate renderer from the Manifest Editor [configuration](../configuration/editor-context).
In case these's no definition for the received annotation type, the component loads an error message at the place of the annotation.
