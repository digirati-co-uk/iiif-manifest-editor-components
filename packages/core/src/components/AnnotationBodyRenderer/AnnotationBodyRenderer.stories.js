import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import AnnotationBodyRenderer from './AnnotationBodyRenderer.js';

storiesOf('AnnotationBodyRenderer', module)
  .addDecorator(withKnobs)
  .add('AnnotationBodyRenderer', () => {
    const annotation = {
      motivation: text('Motivation', 'painting'),
      body: {
        type: text('Type', '3DModel'),
        value: text('Value', 'Value'),
      },
    };
    return <AnnotationBodyRenderer annotation={annotation} />;
  });
