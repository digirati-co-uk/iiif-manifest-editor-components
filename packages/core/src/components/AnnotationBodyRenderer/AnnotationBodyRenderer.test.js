import * as React from 'react';
import AnnotationBodyRenderer from './AnnotationBodyRenderer';

describe("AnnotationBodyRenderer", () => {
  it('Renders NotSupportedAnnotation if the passed annotation is not configured', () => {
    const annotation = {
      motivation: 'painting',
      body: {
        type: '3DModel'
      }
    };
    const renderedBody = render(<AnnotationBodyRenderer annotation={annotation}/>);
    expect(renderedBody).toMatchSnapshot();
  });
  it('Renders NotSupportedAnnotation if the passed annotation exist', () => {
    const annotation = {
      motivation: 'painting',
      body: {
        type: 'TextualBody',
        value: 'test',
      }
    };
    const renderedBody = render(<AnnotationBodyRenderer annotation={annotation}/>);
    expect(renderedBody).toMatchSnapshot();
  });
})