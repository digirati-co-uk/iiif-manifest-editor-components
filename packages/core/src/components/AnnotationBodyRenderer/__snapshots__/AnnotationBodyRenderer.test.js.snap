// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnotationBodyRenderer Renders NotSupportedAnnotation if the passed annotation exist 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "style": "pointer-events:none",
    },
    "children": Array [
      Object {
        "data": "test",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`AnnotationBodyRenderer Renders NotSupportedAnnotation if the passed annotation is not configured 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "Annotation type[3DModel::painting] not supported.",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "p",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
